### ISSUE TITLE
twistd mail needs better usage doc

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5922]
|<img alt="magmatt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| magmatt reported|
|-|-|
|Trac ID|trac#5922|
|Type|enhancement|
|Created|2012-08-27 20:17:51Z|
|Branch|https://github.com/twisted/twisted/tree/mail-tap-usage-5922|

twistd mail --help doesn't make it obvious how domains work or how to get a useful smtp server running.  A simple example would probably help a lot.

Attachments:

* [twistd-mail-usage.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/364/364f5a1d50a658ff06ade29daf12c25c21de114d/d7bb3c10088089fed90a4be98a3ab1e4f380d123.patch) (772 bytes) - added by magmatt on 2012-08-27 20:18:10Z - improved twistd mail doc
* [twistd-mail-usage-5922.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/364/364f5a1d50a658ff06ade29daf12c25c21de114d/bc03fe7dc12e912d8d66db789463e1eb91fc5d15.patch) (1672 bytes) - added by rwall on 2012-09-13 00:14:23Z - A slightly more verbose documentation update

<details><summary>Searchable metadata</summary>

```
trac-id__5922 5922
type__enhancement enhancement
reporter__magmatt magmatt
priority__normal normal
milestone__ 
branch__branches_mail_tap_usage_5922 branches/mail-tap-usage-5922
branch_author__jerub jerub
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__None None
time__1346098671000000 1346098671000000
changetime__1474198049187875 1474198049187875
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_2"></a>|@wallrj commented|
|-|-|

This started as a code review of the first patch, but it was interesting to use the twistd mail plugin for the first time and I have ended up creating a second patch. 

I certainly think that the first patch makes the mail plugin easier to use. 

I have added an Example heading and added an example of running a simple mail relay.

Also added some documentation directly to the --maildirdbmdomain option explaining the format of the required argument.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_3"></a>|@jerub commented|
|-|-|

(In [35759]) New branch for bug [#5922](https://github.com/twisted/twisted/issues/5922).

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_4"></a>|@jerub commented|
|-|-|

(In [35761]) Apply twistd-mail-usage-5922.patch from [[#5922](https://github.com/twisted/twisted/issues/5922)](https://github.com/twisted/twisted/issues/5922) by rwall based on magmatt's
original patch.

Refs #5922

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_5"></a>|@jerub commented|
|-|-|

(In [35765]) Update example commandlines for the 'mail' twistd plugin to use the long format descriptions.

Refs [#5922](https://github.com/twisted/twisted/issues/5922)

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_6"></a>|@jerub set status to `closed`|
|-|-|

(In [35767]) Merge mail-tap-usage-5922: Update documentation on 'mail' twistd plugin

Author: magmatt, rwall
Reviewer: jerub
Fixes: [#5922](https://github.com/twisted/twisted/issues/5922)

twistd mail --help output now gives some helpful example usage.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
